src/workerd/server/workerd-api.c++

```diff
@@ -737,6 +918,14 @@ void WorkerdApi::compileModules(jsg::Lock& lockParam,
       auto bundle = KJ_ASSERT_NONNULL(
           fetchPyodideBundle(impl->pythonConfig, version), "Failed to get Pyodide bundle");
 
+      // Get Python requirements and fetch packages
+      KJ_IF_SOME(a, artifacts) {
+        KJ_IF_SOME(pm, a->packageManager) {
+          auto pythonRequirements = getPythonRequirements(source);
+          fetchPyodidePackages(impl->pythonConfig, pm, pythonRequirements, pythonRelease);
+        }
+      }
```

Comment:
Will also need to add this to the new module registry setup once https://github.com/cloudflare/workerd/pull/4191 lands... If possible I'd like to get #4191 landed first

Reply:
That PR landed now.